src/services/codefixes/fixAddVoidToPromise.ts

```diff
@@ -0,0 +1,82 @@
+/* @internal */
+namespace ts.codefix {
+    const fixName = "addVoidToPromise";
+    const fixId = "addVoidToPromise";
+    const errorCodes = [
+        Diagnostics.Expected_0_arguments_but_got_1_Did_you_forget_to_include_void_in_your_type_argument_to_Promise.code
+    ];
+    registerCodeFix({
+        errorCodes,
+        fixIds: [fixId],
+        getCodeActions(context) {
+            const changes = textChanges.ChangeTracker.with(context, t => makeChange(t, context.sourceFile, context.span, context.program));
+            if (changes.length > 0) {
+                return [createCodeFixAction(fixName, changes, Diagnostics.Add_void_to_Promise_resolved_without_a_value, fixId, Diagnostics.Add_void_to_all_Promises_resolved_without_a_value)];
+            }
+        },
+        getAllCodeActions(context: CodeFixAllContext) {
+            return codeFixAll(context, errorCodes, (changes, diag) => makeChange(changes, diag.file, diag, context.program, new Set()));
+        }
+    });
+
+    function makeChange(changes: textChanges.ChangeTracker, sourceFile: SourceFile, span: TextSpan, program: Program, seen?: Set<ParameterDeclaration>) {
+        const node = getTokenAtPosition(sourceFile, span.start);
+        if (!isIdentifier(node) || !isCallExpression(node.parent) || node.parent.expression !== node || node.parent.arguments.length !== 0) return;
+
+        const checker = program.getTypeChecker();
+        const symbol = checker.getSymbolAtLocation(node);
+
+        // decl should be `new Promise((<decl>) => {})`
+        const decl = symbol?.valueDeclaration;
+        if (!decl || !isParameter(decl) || !isNewExpression(decl.parent.parent)) return;
+
+        // no need to make this change if we have already seen this parameter.
+        if (seen?.has(decl)) return;
+        seen?.add(decl);
+
+        const typeArguments = getEffectiveTypeArguments(decl.parent.parent);
+        if (some(typeArguments)) {
+            // append ` | void` to type argument
+            const typeArgument = typeArguments[0];
+            const needsParens = !isUnionTypeNode(typeArgument) && !isParenthesizedTypeNode(typeArgument) &&
+                isParenthesizedTypeNode(factory.createUnionTypeNode([typeArgument, factory.createKeywordTypeNode(SyntaxKind.VoidKeyword)]).types[0]);
+            if (needsParens) {
+                changes.insertText(sourceFile, typeArgument.pos, "(");
+            }
+            changes.insertText(sourceFile, typeArgument.end, needsParens ? ") | void" : " | void");
+        }
+        else {
+            // make sure the Promise is type is untyped (i.e., `unknown`)

```

Comment:
Wasn't this checked when the diagnostic was produced?

Reply:
Not quite. This error can be produced from the following case, but we can't apply a quick fix to it:

```ts
interface Obj {
  m(): Promise<number>;
}
const obj: Obj = {
 m: () => new Promise(resolve => resolve());
}
```
We don't check the `type` of the promise in the checker, just the arity of the call to `resolve`. Adding `void` to `new Promise` would result in an error, and I'm not quick-fixing the definition of `m` on the `Obj` interface, as it could be code you don't own (such as from an external library), or could have repercussions throughout the codebase as you introduce a `void` that wasn't previously there. Instead, I check whether the  promise is untyped (`unknown` in TS, `any` in JS).